test/tools/jsontests/LegacyTestsBoost.cpp

```diff
@@ -0,0 +1,220 @@
+#include <test/tools/jsontests/BlockChainTests.h>
+#include <test/tools/jsontests/StateTests.h>
+#include <test/tools/libtesteth/TestHelper.h>
+
+#include <boost/test/unit_test.hpp>
+
+using namespace dev::test;
+namespace fs = boost::filesystem;
+
+fs::path StateTestSuiteLegacyConstantinople::suiteFolder() const
+{
+    if (Options::get().fillchain || Options::get().filltests)
+        BOOST_FAIL("Legacy tests are sealed and not refillable!");
+    return "LegacyTests/Constantinople/GeneralStateTests";
+}
+
+fs::path StateTestSuiteLegacyConstantinople::suiteFillerFolder() const
+{
+    return "LegacyTests/Constantinople/GeneralStateTestsFiller";
+}
+
+fs::path BCGeneralStateTestsSuiteLegacyConstantinople::suiteFolder() const
+{
+    if (Options::get().filltests)
+        BOOST_FAIL("Legacy tests are sealed and not refillable!");
+    return "LegacyTests/Constantinople/BlockchainTests/GeneralStateTests";
+}
+
+fs::path BCGeneralStateTestsSuiteLegacyConstantinople::suiteFillerFolder() const
+{
+    return "LegacyTests/Constantinople/BlockchainTestsFiller/GeneralStateTests";
+}
+
+
+class LegacyConstantinopleGeneralStateTestFixture
+{
+public:
+    LegacyConstantinopleGeneralStateTestFixture()
+    {
+        StateTestSuiteLegacyConstantinople suite;
+        string casename = boost::unit_test::framework::current_test_case().p_name;
+        boost::filesystem::path suiteFillerPath = suite.getFullPathFiller(casename).parent_path();
+        if (!test::Options::get().all)
+        {
+            std::cout << "Skipping " << casename << " because --all option is not specified.\n";
+            test::TestOutputHelper::get().markTestFolderAsFinished(suiteFillerPath, casename);
+            return;
+        }
+        suite.runAllTestsInFolder(casename);
+        test::TestOutputHelper::get().markTestFolderAsFinished(suiteFillerPath, casename);
+    }
+};
+
+class LegacyConstantinopleBCGeneralStateTestFixture
+{
+public:
+    LegacyConstantinopleBCGeneralStateTestFixture()
+    {
+        BCGeneralStateTestsSuiteLegacyConstantinople suite;
+        string const casename = boost::unit_test::framework::current_test_case().p_name;
+        boost::filesystem::path suiteFillerPath = suite.getFullPathFiller(casename).parent_path();
+
+        // skip this test suite if not run with --all flag (cases are already tested in state tests)
+        if (!test::Options::get().all)
+        {
+            cnote << "Skipping hive test " << casename << ". Use --all to run it.\n";
+            test::TestOutputHelper::get().markTestFolderAsFinished(suiteFillerPath, casename);
+            return;
+        }
+        suite.runAllTestsInFolder(casename);
+        test::TestOutputHelper::get().markTestFolderAsFinished(suiteFillerPath, casename);
+    }
+};
+
+
+BOOST_AUTO_TEST_SUITE(LegacyTests)
+    BOOST_AUTO_TEST_SUITE(Constantinople)
```

Comment:
Are these testing only Constantinople?

Reply:
Ah I see it repeats tests repo structure. Why is the subdir called Constantinople then?

Reply:
Because it's Constantinople tests snapshot. Meaning all the tests we had at the moment of Constantinople. It actually includes ConstantinopleFix also which I count as Constantinople as well.